torch/sparse/semi_structured.py

```diff
@@ -320,70 +314,25 @@ def __torch_dispatch__(cls, func, types, args, kwargs) -> Any:
             metadata = args[0].compressed_tensor[num_kept_elements:].view(m, -1)
 
             # the metadata is expected to be in different datatypes for fp16/int8 respectively for CUTLASS.
-            if args[0].dtype is torch.int8:
-                return metadata.view(torch.int32)
-            elif args[0].dtype is torch.float16:
-                return metadata.view(torch.int16)
+            indices_dtype = SparseSemiStructuredTensor.__get_indices_dtype(args[0].dtype)
+            return metadata.view(indices_dtype)
 
         error_string = "\n".join(
             [f"func {func} with args: "]
             + [f"arg{i}: {arg}" for i, arg in enumerate(args)]
         )
         raise NotImplementedError(error_string)
 
+    def to_dense(self):
+        if self.compressed_tensor is None:
+            raise RuntimeError("Compressed tensor is not set, cannot convert to dense!")
 
-def to_sparse_semi_structured(
```

Comment:
Added back.  I've changed calls to this function in test and benchmark scripts to calling `SparseSemiStructuredTensor` constructor, would these be preferable reverted too?

Reply:
I think it might be good here for the `to_sparse_semi_structured` to act as a factory for the `SparseSemiStructuredTensor` subclass.  Is that what you mean @jcaip?

Reply:
Yes, that's right - @alexsamardzic yup please restore those as well.